src/librbd/internal.cc

```diff
@@ -1318,12 +1318,237 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
 
     C_SaferCond cond;
     auto req = librbd::image::RemoveRequest<>::create(
-      io_ctx, image_name, image_id, force, prog_ctx, op_work_queue, &cond);
+      io_ctx, image_name, image_id, force, from_trash_remove, prog_ctx,
+      op_work_queue, &cond);
     req->send();
 
     return cond.wait();
   }
 
+  int trash_move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
+                 const std::string &image_name, uint64_t delay) {
+    CephContext *cct((CephContext *)io_ctx.cct());
+    ldout(cct, 20) << "trash_move " << &io_ctx << " " << image_name
+                   << dendl;
+
+    std::string image_id;
+    ImageCtx *ictx = new ImageCtx(image_name, "", nullptr, io_ctx, false);
```

Comment:
@rjfd No worries -- I already have a patch for it. I'm just adding a new API so that ```rbd info``` and return some text that the parent is actually deleted.